### PR TITLE
chore: refactor init scripts

### LINKED ISSUE #1: chore: refactor init scripts
### Tasks

Add a pre-commit script that can:
  - [ ] Install Python for those who don't have it, and differentiate between Windows dan Ubuntu.
  - [ ] Find how to install dependencies in Windows.
  - [ ] Find how to check if dependencies are already installed. (probably use aliases)

### PR DESCRIPTION
**Pull request by VioAlbert**
In this PR, the file `init.sh` and `init.bat` are adjusted according to the needs in issue #1.

### CODE REVIEW

File: init.bat [hunk 1]
```diff
@@ -1,27 +1,39 @@
...
-    set /p install_python=Python is not installed. Do you want to install it? (Y/N)
-    if /i "%install_python%"=="Y" (
-        echo Installing Python...
-        powershell -Command "& {$ProgressPreference='SilentlyContinue'; Invoke-WebRequest '%PYTHON_URL%' -OutFile 'python-%PYTHON_VERSION%-amd64.exe'}"
-        start /wait python-%PYTHON_VERSION%-amd64.exe /quiet
-        del python-%PYTHON_VERSION%-amd64.exe
-        echo Python has been installed.
-    )
+  :installpython
+  set /p install_python="Python %PYTHON_VERSION:~0,3% is not installed. Do you want to install it? (Y/N)"
```
Comment (stevennovaryo) [inline init.bat]: Hmm probably it is better to mention that we will download python in this step.

File: init.sh [hunk 2]
```diff
@@ -40,31 +40,52 @@ installPreCommit() {
...
       [Nn]* ) break;;
-      * ) echo "Please answer yes or no.";;
+      * ) ;;
   esac
 done
+
+installTcrand() {
+  run_cmd "git clone https://github.com/afaji/tcrand $HOME/tcrand"
+  run_cmd "cp -r $HOME/tcrand/include/tcrand $HOME/tcframe/include"
+  need_reload=1
```
Comment (stevennovaryo) [inline init.sh]: Do the user need to reload the terminal before using tcrand? If i'm not wrong the c++ spec will compile just fine?

File: init.sh [hunk 3]
```diff
@@ -40,31 +40,52 @@ installPreCommit() {
 }
 
 while true; do
-  read -p "Do you want to install precommit? (Y/n) " yn
+  read -p "Do you want to install pre-commit? (Y/n) " yn
```
Comment (stevennovaryo) [inline init.sh]: Maybe rename it to python and pre-commit, i forgot about it too.

**Review by stevennovaryo — CHANGES_REQUESTED** [review]

File: init.bat (hunk 1 above)
Comment (VioAlbert) [reply init.bat]: Will the prompt `Checking python {VERSION} installation...` suffice?

File: init.sh (hunk 2 above)
Comment (VioAlbert) [reply init.sh]: This is an overlook on my part, will be fixed.

File: init.sh (hunk 3 above)
Comment (VioAlbert) [reply init.sh]: Noted.

File: init.bat (hunk 1 above)
Comment (stevennovaryo) [reply init.bat]: Oh my concern is that python size is quite big and it may burden some users. What do you think?

Comment (VioAlbert) [reply init.bat]: Discussed via Discord. Will be fixed ASAP.

**Review by stevennovaryo — APPROVED** [review]

**Pull request merged**